docs/tutorials/custom.rst

```diff
@@ -6,58 +6,111 @@ While Flambé offers a large number of `Component` objects to use in experiments
-such as ``flambe.nn.Module`` or ``flambe.nn.Dataset``. Alternatively you can also inherit from
-the ``flambe.Component`` object directly.
+.. code-block:: python
 
-A Component must implement a ``run`` method which returns a boolean indicating whether execution
-should continue or not (useful for multi-step components such as a ``Trainer``).
+    from setuptools import setup, find_packages
 
-.. attention:: Make sure that all your components are surfaced at the top level __init__, as This
-                is how Flambé will register them for config usage.
+    setup(
+        name = 'my_package', # the name of your package
+        version = '1.0.0', # the version of your extension (optional)
+        packages = find_packages(),
+        install_requires = ['numpy >= 1.11.1', 'matplotlib >= 1.5.1'], # Idicate dependencies here
+    )
 
-You have now built your first extension! You can now use it freely in any configuration,
-whether that'd be for an ``Experiment``, a ``Cluster`` or any other ``Runnable``:
+After you add the setup file to your package, your final folder structure should look like
+the one below:
 
+::
+
+    my_package
+    ├── setup.py # This file makes your package installable
+    └── my_module
+        ├── __init__.py
+        ├── model.py
+        ├── dataset.py
+        └── ...
+
+
+Using your extension
+--------------------
+
+You have built your first extension! You can now use it freely in any configuration,
+whether that'd be for an ``Experiment``, a ``Cluster`` or any other ``Runnable``.
+
+To do so, simply add them at the top of your extension, mapping the name of the
+module your built (``my_module`` in the example) to the location of the package
+(``my_package`` in the example).
+
+.. important:: The name of the module is used as a prefix in your configurations. Not
+               the name of your package.
 
 .. code-block:: yaml
 
-    my_project: /path/to/my_pip_installable/  # Must map from module to package location
-    ---
+    my_module: path/to/my_package  # Must map from module to package path
+
+    ---  # Note the 3 dashes here
 
-    Experiment
+    !Experiment
 
     pipeline:
-        dataset: !my_project.MyDataset # We use the name of your custom module as prefix
+        dataset: !my_module.MyDataset # We use the name of your custom module as prefix
 
 .. tip:: The path to the package may be a local path, a github URL, or the name of package one
-          pipy. The latter allows you to specify a specific version of your extension. For github,
+          pypi. The latter allows you to specify a specific version of your extension. For github,
           we also support links to specific commit or branches.
+
+Flambé will require your extension to be installed. You can do so manually by running:
+
+``pip install my_package`` 
+
+or Flambé can install all the extensions specified in your
+configuration automatically if the ``-i`` flag is used:
```

Comment:
Perhaps we say: "...automatically _while executing your config_ if the..."
to make it super clear the `-i` flag is just adding something to normal execution and we'll still be executing the config.

Reply:
Will add that!